### ISSUE TITLE
unused argument (group_scale = scalevals)

### ISSUE THREAD

**Issue by marcosci** [tylermorganwall/rayshader#197]
Hi Tyler,

can't figure this one us by myself, as the code worked yesterday.
Only difference: new macOs version.

The error seems to come from rayrender, but only happens with render_highquality.

Any idea?

Cheers

``` r
install.packages('rayrender', repos = 'https://tylermorganwall.r-universe.dev')
#> 
#> The downloaded binary packages are in
#>  /var/folders/w7/4n7x5s5s7ydf7l095hhvg80h0000gn/T//RtmpSzO6mS/downloaded_packages
install.packages('rayshader', repos = 'https://tylermorganwall.r-universe.dev')
#> 
#> The downloaded binary packages are in
#>  /var/folders/w7/4n7x5s5s7ydf7l095hhvg80h0000gn/T//RtmpSzO6mS/downloaded_packages

library(sf)
#> Linking to GEOS 3.8.1, GDAL 3.2.1, PROJ 7.2.1
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(fasterize)
#> 
#> Attaching package: 'fasterize'
#> The following object is masked from 'package:graphics':
#> 
#>     plot
#> The following object is masked from 'package:base':
#> 
#>     plot
library(rayshader)
library(raster)
#> Loading required package: sp
#> 
#> Attaching package: 'raster'
#> The following object is masked from 'package:dplyr':
#> 
#>     select
library(exactextractr)
library(rayrender)
#> Warning: package 'rayrender' was built under R version 4.1.2

library(rnaturalearth)
borders  <- ne_countries(country = "Germany", returnclass = "sf") 

rivers <- ne_download(scale = 10, 
                        type = 'rivers_lake_centerlines', 
                        category = 'physical', 
                        returnclass = "sf") |> 
  st_intersection(borders) |> 
  st_cast("MULTIPOINT")
#> OGR data source with driver: ESRI Shapefile 
#> Source: "/private/var/folders/w7/4n7x5s5s7ydf7l095hhvg80h0000gn/T/RtmpSzO6mS", layer: "ne_10m_rivers_lake_centerlines"
#> with 1455 features
#> It has 34 fields
#> Integer64 fields read as strings:  rivernum ne_id
#> Warning in rgdal::readOGR(destdir, file_name, encoding = "UTF-8",
#> stringsAsFactors = FALSE, : Dropping null geometries: 1453
#> Warning: attribute variables are assumed to be spatially constant throughout all
#> geometries

distance_to_water <- distanceFromPoints(raster(as(borders, "Spatial"), ncols=250, nrows=170),
                           st_coordinates(rivers)[,1:2])


distance_to_water <- mask(distance_to_water, borders)

# make hexagonal grid
hexbin_vec <- st_make_grid(st_as_sfc(st_bbox(distance_to_water)), cellsize = .1, square = FALSE)

# get mean raster values for each hexagon
values <- exact_extract(distance_to_water, hexbin_vec, "mean")
#> Warning in .local(x, y, ...): Polygons transformed to raster CRS (EPSG:NA)
#>   |                                                                              |                                                                      |   0%  |                                                                              |                                                                      |   1%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |==                                                                    |   4%  |                                                                              |===                                                                   |   4%  |                                                                              |===                                                                   |   5%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |=====                                                                 |   8%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |=======                                                               |  11%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |=========                                                             |  14%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  16%  |                                                                              |============                                                          |  17%  |                                                                              |============                                                          |  18%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |==============                                                        |  21%  |                                                                              |===============                                                       |  21%  |                                                                              |===============                                                       |  22%  |                                                                              |================                                                      |  22%  |                                                                              |================                                                      |  23%  |                                                                              |================                                                      |  24%  |                                                                              |=================                                                     |  24%  |                                                                              |=================                                                     |  25%  |                                                                              |==================                                                    |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |===================                                                   |  28%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |=====================                                                 |  31%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |=======================                                               |  34%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |==========================                                            |  38%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  39%  |                                                                              |============================                                          |  40%  |                                                                              |============================                                          |  41%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |==============================                                        |  44%  |                                                                              |===============================                                       |  44%  |                                                                              |===============================                                       |  45%  |                                                                              |================================                                      |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |=================================                                     |  48%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |===================================                                   |  51%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |=====================================                                 |  54%  |                                                                              |======================================                                |  54%  |                                                                              |======================================                                |  55%  |                                                                              |=======================================                               |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  56%  |                                                                              |========================================                              |  57%  |                                                                              |========================================                              |  58%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |==========================================                            |  61%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  62%  |                                                                              |============================================                          |  63%  |                                                                              |============================================                          |  64%  |                                                                              |=============================================                         |  64%  |                                                                              |=============================================                         |  65%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |===============================================                       |  68%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |=================================================                     |  71%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |===================================================                   |  74%  |                                                                              |====================================================                  |  74%  |                                                                              |====================================================                  |  75%  |                                                                              |=====================================================                 |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  76%  |                                                                              |======================================================                |  77%  |                                                                              |======================================================                |  78%  |                                                                              |=======================================================               |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  79%  |                                                                              |========================================================              |  80%  |                                                                              |========================================================              |  81%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |==========================================================            |  84%  |                                                                              |===========================================================           |  84%  |                                                                              |===========================================================           |  85%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |=============================================================         |  88%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |===============================================================       |  91%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |=================================================================     |  94%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |====================================================================  |  98%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================|  99%  |                                                                              |======================================================================| 100%

# turn it into a simple feature and give it values
hexbin_vec <- st_sf(geom=hexbin_vec)
hexbin_vec$distance <- values

# turn hexbin into raster
hexbins <- fasterize(hexbin_vec,
                     distance_to_water,
                     field = "distance",
                     fun="sum")

# and the raster into a rayrender!
elevation_matrix <- raster_to_matrix(hexbins)

elevation_matrix |>
  height_shade() |>
  add_shadow(ray_shade(elevation_matrix, zscale=2, sunaltitude=25, sunangle=315), 0.3) |>
  plot_3d(elevation_matrix, 
          zscale = 5000,
          shadow = TRUE,
          fov = 35, 
          theta = 0, 
          phi = 40,
          asp = 1.6,
          zoom = 0.7)

render_highquality(filename = "distance_highres.png", 
                   samples=500, 
                   lightdirection = c(60,120, 240),
                   lightaltitude=c(90,25, 12),
                   lightintensity=c(100, 500, 450),
                   lightcolor = c("white", "#FF9956", "#FF79E7"),
                   progress=TRUE,
                   parallel=TRUE)
#> Error in rayrender::group_objects(scene, group_scale = scalevals, pivot_point = c(0, : unused argument (group_scale = scalevals)
```

<sup>Created on 2021-11-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by tylermorganwall** [comment]
Thanks! This should be fixed in the latest version (b74dd271e12b047bc9f790f4fae248c52171e26f).

**Comment by marcosci** [comment]
It is, sweet! Thanks a ton, 30DayMapChallenge can go on :D
